lib/modules/datasource/docker/index.ts

```diff
@@ -824,41 +824,86 @@ export class DockerDatasource extends Datasource {
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        const manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentValue
+        );
+        if (manifestResponse) {
+          const manifestList = JSON.parse(manifestResponse.body) as
+            | ImageList
+            | OciImageList;
+          for (const manifest of manifestList.manifests) {
+            if (manifest.digest === currentDigest) {
+              architecture =
+                (manifest.platform['architecture'] as string) || null;
+              logger.debug(
+                `Current digest ${currentDigest} relates to architecture ${
+                  architecture ?? 'null'
+                }`
+              );
+              break;
+            }
+          }
```

Comment:
you need to validate that's a manifest list, it can also be a simple image. Check `getConfigDigest` function for sample how to differenciate

```suggestion
          const manifestList = JSON.parse(manifestResponse.body) as
            | ImageList
            | Image
            | OciImageList
            | OciImage;
          if(manifest.schemaVersion === 2 && manifest.mediaType === MediaType.manifestListV2) {
            for (const manifest of manifestList.manifests) {
              if (manifest.digest === currentDigest) {
                architecture =
                  (manifest.platform['architecture'] as string) ?? null;
                logger.debug(
                  `Current digest ${currentDigest} relates to architecture ${
                    architecture ?? 'null'
                  }`
                );
                break;
              }
            }
          }
```